packages/tizen-remote/test/e2e/tizen-remote.e2e.spec.js

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable promise/no-native */
+
+/**
+ * This test suite checks the interaction of the lib with a WebSocket server.
+ *
+ * If the following env variables are present, the test suite will use them:
+ *
+ * - `TEST_TIZEN_REMOTE_HOST`: The host of the WebSocket server.
+ * - `TEST_TIZEN_REMOTE_PORT`: The port of the WebSocket server.
+ *
+ * If these env variables are _not_ present, a mock server will be started on a random port.
+ *
+ * @module
+ */
+
+import {createSandbox} from 'sinon';
+import getPort from 'get-port';
+import {TestWSServer} from './server';
+import unexpected from 'unexpected';
```

Comment:
would be nice if this lib exported assertions so you don't have to remember strings

Reply:
it's the diff between remembering strings or remembering the pattern of ways chai abuses object properties

anyway, I prefer this lib over chai due to that and:

1. it is much easier to write custom assertions. I wrote [these](https://github.com/mochajs/mocha/blob/master/test/assertions.js) for mocha. plugins (like [unexpected-eventemitter](https://npm.im/unexpected-eventemitter) are just collections of assertions
2. the DSL for the "string assertions" isn't that scary and is a reasonable use of a DSL
3. built-in promise support
4. better diffs than literally anything else I've seen
5. despite being relatively unpopular, has a nice ecosystem

the disadvantages for me:

1. basically nobody is familiar with it and it looks weird at first
2. no TS declarations
3. there is some black magic in it which I am wary of (though have not yet been bitten by)

Reply:
ok, that's cool. for IDE discoverability you'd think they would export assertion constants or something that just are the correct strings.

Reply:
might be a little bit awkward to name those constants, but certainly type declarations could be created containing those string literals.  that said, unless those constants are verbose, semantic meaning will likely get lost.